### PR TITLE
[#74] res.cookie & res.clearCookie express style overloads

### LINKED ISSUE #74: [FEAT] Support Express style signature for cookie APIs
# Issue

Support the Express signature for the `cookie()` and `clearCookie()` APIs.

These APIs already exist in Opine, but the signature is different - expecting a full `Cookie` object vs the Express name, value and options signature.

## Details

See:

- https://expressjs.com/en/api.html#res.cookie
- https://expressjs.com/en/api.html#res.clearCookie

### PR DESCRIPTION
**Pull request by FaberVitale**
# [FEAT] Support Express style signature for cookie APIs

[#74](https://github.com/asos-craigmorten/opine/issues/74)

## Details

Implements express styles cookie api and closes #74

## CheckList

- [x] PR starts with [#74].
- [x] Has been tested (where required) before merge to main.

### CODE REVIEW

File: test/units/res.cookie.test.ts [hunk 1]
```diff
@@ -17,6 +17,32 @@ describe("res", function () {
         .expect(200, done);
     });
 
+    it("should raise an exception if no arg is provided", function (done) {
+      const app = opine();
+
+      app.use(function (req, res) {
+        (res as any).cookie().end();
```
Comment (asos-craigmorten) [inline test/units/res.cookie.test.ts:24]: Assuming the typing is working correctly I would hope this `any` casting shoudn't be required (as per previous tests).

```suggestion
        res.cookie().end();
```

File: test/units/res.cookie.test.ts [hunk 2]
```diff
@@ -17,6 +17,32 @@ describe("res", function () {
...
+        .expect(500, done);
+    });
+
+    it("should raise an expection if args provided are not valid", function (
+      done,
+    ) {
+      const app = opine();
+
+      app.use(function (req, res) {
+        (res as any).cookie({}).end();
```
Comment (asos-craigmorten) [inline test/units/res.cookie.test.ts:38]:
```suggestion
        res.cookie({}).end();
```

File: test/units/res.clearCookie.test.ts [hunk 3]
```diff
@@ -1,9 +1,21 @@
 import { opine } from "../../mod.ts";
-import { superdeno } from "../deps.ts";
+import { superdeno, SuperDenoResponse, expect } from "../deps.ts";
```
Comment (asos-craigmorten) [inline test/units/res.clearCookie.test.ts:2]: If `SuperDenoResponse` is indeed required (would hope that we get the types for free - shout it not and can investigate in the SuperDeno repo!) we should import using the `import type` syntax:

```suggestion
import { superdeno, expect } from "../deps.ts";
import { SuperDenoResponse } from "../deps.ts";
```

**Comment by asos-craigmorten** [comment]
One fix r.e. types in the test (`import type` becoming a thing in Deno 1.4.0), otherwise this looks fab @FaberVitale! Nice One! 💯

**Comment by FaberVitale** [comment]
Apparently there has been a change from 1.4.1 to 1.4.3 to `deno fmt` and `deno lock`.

I had to regenerate the lock file and fmt has changed the order of named imports.

**Review by asos-craigmorten — APPROVED** [review]
Looks fab @FaberVitale - nice one.

Curious around the formatter! Alphabetized imports feels like another borrow from goland perhaps 🤔 

:shipit:

**Pull request merged**